inference-engine/src/gna_plugin/transformations/decompose_mvn.cpp

```diff
@@ -0,0 +1,261 @@
+
+    // Add epsilon inside the square root
+    auto add_epsilon = std::make_shared<opset7::Add>(eps_tensor_const, reshape_avg_conv_4);
+
+    // Calculate square root and inversion
+    auto log_var_eps = std::make_shared<opset7::Log>(add_epsilon);
+    log_var_eps->set_friendly_name("MvnLogVarEps");
+    auto log_inv_stdev = std::make_shared<opset7::Multiply>(log_var_eps, minus_half_const);
+    log_inv_stdev->set_friendly_name("MvnLogInvStdev");
+    inv_stdev = std::make_shared<opset7::Exp>(log_inv_stdev);
+    inv_stdev->set_friendly_name("MvnInvStdev");
+    copy_runtime_info(mvn, {add_epsilon, log_var_eps, log_inv_stdev, inv_stdev});
+
+    auto normalized_output = std::make_shared<opset7::Multiply>(subtract_mean, inv_stdev);
+    normalized_output->set_friendly_name("MvnOutput");
+
+    copy_runtime_info(mvn, {squared_diff, squared_diff_reshape, transposed_input_3, transposed_avg_conv_3, avg_conv_3, reshape_avg_conv_3,
+        transposed_input_4, transposed_avg_conv_4, avg_conv_4, reshape_avg_conv_4});
+
+    return normalized_output;
+}
+
+static void Decompose(const std::shared_ptr<opset7::MVN> mvn, const MVNData& mvn_data) {
+    // Prepare data
+    auto combined_C_H = mvn_data.C * mvn_data.H;
+
+    std::vector<float> neg_avg_weights(8 * mvn_data.W / mvn_data.num_parts, -1.0f / mvn_data.W);
+    auto neg_avg_weights_const = opset7::Constant::create(mvn_data.element_type, Shape{8, mvn_data.W / mvn_data.num_parts, 1, 1}, neg_avg_weights);
+
+    std::vector<float> avg_broadcast(8 * mvn_data.W * mvn_data.num_parts, 0.0f);
+    for (size_t i = 0; i < mvn_data.W * mvn_data.num_parts; i++) {
+        avg_broadcast[i * 8] = 1.0f;
+    }
+    auto avg_broadcast_const = opset7::Constant::create(mvn_data.element_type, Shape{mvn_data.W, 8 * mvn_data.num_parts, 1, 1}, avg_broadcast);
+
+    // Create average calculation part of the graph
+    // We assume C = 1 case (combined channels)
+    const auto input = mvn->input_value(0);
+    auto reshape = std::make_shared<opset7::Reshape>(input,
+        opset7::Constant::create(element::i64, Shape{4}, Shape{mvn_data.N, 1ull, combined_C_H, mvn_data.W}), false);
+    auto input_4d = std::make_shared<opset7::Reshape>(reshape,
+        opset7::Constant::create(element::i64, Shape{4}, Shape{mvn_data.N, combined_C_H * mvn_data.num_parts, 1ull, mvn_data.W / mvn_data.num_parts}), false);
+    auto input_2d = std::make_shared<opset7::Reshape>(reshape,
+        opset7::Constant::create(element::i64, Shape{2}, Shape{1ull, combined_C_H * mvn_data.W}), false);
+    auto transposed_input_1 = std::make_shared<opset7::Transpose>(input_4d, opset7::Constant::create(element::i64, Shape{4}, {0, 3, 1, 2}));
+    auto transposed_avg_conv_1 = std::make_shared<opset7::Convolution>(transposed_input_1, neg_avg_weights_const,
+        Strides{1, 1}, CoordinateDiff{0, 0}, CoordinateDiff{0, 0}, Strides{1, 1}, op::PadType::VALID);
+    transposed_avg_conv_1->set_friendly_name("MvnAvg1");
```

Comment:
Names of new operations should be related to operation name which you replace.

Reply:
I think they fit here well, because all the new operations replace only one original layer: MVN. And the new operation names follow and describe the purpose of the operation.